src/utils/DatabaseFilter.cpp

```diff
@@ -0,0 +1,196 @@
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Oliver Alka $
+// $Authors: Oliver Alka $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/FORMAT/FileHandler.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/FASTAFile.h>
+#include <OpenMS/FORMAT/MzIdentMLFile.h>
+#include <OpenMS/FORMAT/FileTypes.h>
+#include <OpenMS/METADATA/PeptideIdentification.h>
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+    @page UTILS_DatabaseFilter DatabaseFilter
+
+    @brief The DatabaseFilter tool filters a protein database in fasta format according to one or multiple filtering criteria.
+
+    The resulting database is written as output. Depending on the reporting method (method="whitelist" or "blacklist") only entries are retained that passed all filters ("whitelist) or failed at least one filter ("blacklist").
+
+    Implemented filter criteria:
+
+        accession: Filter database according to the set of protein accessions contained in an identification file (idXML, mzIdentML)
+
+    <B>The command line parameters of this tool are:</B>
+    @verbinclude UTILS_DatabaseFilter.cli
+    <B>INI file documentation of this tool:</B>
+    @htmlinclude UTILS_DatabaseFilter.html
+*/
+
+// We do not want this class to show up in the docu:
+/// @cond TOPPCLASSES
+
+class TOPPDatabaseFilter :
+  public TOPPBase
+{
+public:
+  TOPPDatabaseFilter() :
+    TOPPBase("DatabaseFilter", "The DatabaseFilter tool filters a protein database in fasta format according to one or multiple filtering criteria", false)
```

Comment:
There's currently only one criterion, so this is a bit misleading/vague. Also, no need to include the tool name in the description. Maybe just:
"Filters a protein database (FASTA format) based on identified proteins"